lib/certificate_chain_serializer.rb

```diff
@@ -0,0 +1,17 @@
+class CertificateChainSerializer
+  DELIMITER = "\n-----END CERTIFICATE-----\n".freeze
+
+  def self.load(chain)
+    return [] unless chain
+    certs = chain.split(DELIMITER)
+    certs.map do |cert|
+      cert += DELIMITER
```

Comment:
can you please explain what are you trying to do here? I am getting a redundant end chain 
```rb
v = Gem::Package.new(File.join("test", "gems", "valid_signature-0.0.0.gem")).spec
CertificateChainSerializer.load(CertificateChainSerializer.dump(v.cert_chain))
...
-----END CERTIFICATE-----
-----END CERTIFICATE-----
```

Reply:
I replaced it with a better implementation using `String#scan` that gets rid of the problem with the end delimiter gettin g stripped out…